code/modules/mob/living/living.dm

```diff
@@ -1636,10 +1628,18 @@
 	UnregisterSignal(src, COMSIG_MOVABLE_MOVED)
 
 
-/mob/living/set_stat(new_stat)
+/mob/living/set_stat()
 	. = ..()
 	if(isnull(.))
 		return
+
+	// If our stat changed, update whether or not we can succumb
+	if (. != stat)
+		if (CAN_SUCCUMB(src))
+			throw_alert("succumb", /obj/screen/alert/succumb)
+		else
+			clear_alert("succumb")
```

Comment:
Put this into the `switch`
```
switch(.) //Previous stat.
	if(DEAD)
		if (CAN_SUCCUMB(src))
			throw_alert("succumb", /obj/screen/alert/succumb)

switch(stat) //Current stat.
	if(DEAD)
		clear_alert("succumb")
```

Reply:
This would be a lot of repeated code, since you're able to succumb in multiple statuses, and can sometimes not succumb while in crit anyway (NODEATH trait).

Reply:
You can only succumb while `UNCONSCIOUS`, no?
Is there any way to succumb while being `DEAD`, `SOFT_CRIT` or `CONSCIOUS`?

Reply:
I don't think so, maybe I'm overthinking something.

Reply:
Actually what you need to do is place this when the `CAN_SUCCUMB` conditions change.
That is, when gaining or losing `TRAIT_NODEATH`, checking for critical condition, and for when going into and out of critical condition.

Reply:
The trait part is somewhat easy, there's some prior code in init_signals.dm.

The hard part is that when you become unconscious, `InCritical` isn't necessarily true, so it won't show you the succumb prompt.

Reply:
That's true. The answer is to refactor how `InCritical` works so it reacts to events.
I think that adding a new `stat` value, `HARD_CRIT`, would be the way to go.

Reply:
That feels like something that would be out of scope for this PR.

Reply:
It's no small request, that is true. Can you cover all event cases of being in crit without doing it, however?

Reply:
I'm not sure what that would entail if not what there is now/used to be with something like `updatehealth`?

Reply:
I guess you'd have to use your old method, yea, of checking health every single time.
Eh, whatever, do as you'd rather. If I find time for it I'll refactor it later.